### PR TITLE
Add inflationary rewards to the Voting contract.

### LINKED ISSUE #405: v1 Oracle voters can retrieve rewards

### PR DESCRIPTION
**Pull request by mrice32**
Fixes #405.

### CODE REVIEW

File: core/contracts/Voting.sol [hunk 1]
```diff
@@ -148,7 +173,12 @@ contract Voting is Testable, MultiRole, OracleInterface {
             "This (time, identifier) pair is not being voted on this round");
 
         VoteInstance storage voteInstance = priceResolution.votes[currentRoundId];
-        voteInstance.committedHashes[msg.sender] = hash;
+        voteInstance.voteSubmissions[msg.sender].committedHash = hash;
+
+        if (votersLastRound[msg.sender] != currentRoundId) {
+            retrieveRewards();
+            votersLastRound[msg.sender] = currentRoundId;
```
Comment (rawfalafel) [inline core/contracts/Voting.sol]: Seems like this should be set after the reveal, not after the commit.

Comment (mrice32) [reply core/contracts/Voting.sol]: Why do you say that?

Comment (rawfalafel) [reply core/contracts/Voting.sol]: Nevermind, I was unsure if this handled the case where a voter commit but doesn't reveal during a round, but looks like its handled.

**Comment by rawfalafel** [comment]
@ptare brought up a better way that's O(1) to insert and O(log(n) to retrieve. Just insert the inflation rate, not the delta, with the roundId. Retrieval's a binary search on the sorted array.

**Comment by mrice32** [comment]
SG, definitely something we can add if we deem that it's warranted (I think we should do a gas cost analysis to verify that the lookup-add gas tradeoff is worthwhile). We may be able to take advantage of [this library](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/8545c99fb106636c194da739bd0ede43a9595580/contracts/utils/Arrays.sol#L18) or a slightly modified form of it that takes in a different type.

**Pull request merged**